jib-plugins-common/src/main/java/com/google/cloud/tools/jib/plugins/common/PluginConfigurationProcessor.java

```diff
@@ -632,9 +632,19 @@ static JavaContainerBuilder getJavaContainerBuilderWithBaseImage(
     }
 
     String classpathString = String.join(":", classpath);
-    String mainClass =
-        MainClassResolver.resolveMainClass(
-            rawConfiguration.getMainClass().orElse(null), projectProperties);
+    String mainClass;
+    try {
+      mainClass =
+          MainClassResolver.resolveMainClass(
+              rawConfiguration.getMainClass().orElse(null), projectProperties);
+    } catch (MainClassInferenceException ex) {
+      if (entrypointDefined) {
+        // We will use the user-given entrypoint, so don't fail.
+        mainClass = "could-not-infer-a-main-class";
```

Comment:
Just a thought, can we call `MainClassResolver#resolveMainClass` only if the entrypoint is not defined?

Reply:
If you remember, we always create the arg files whether or not an entrypoint is defined so that people can use them in their custom entrypoint.

Reply:
Oh right, sorry that makes sense.